router/batchrouter/partition_worker.go

```diff
@@ -0,0 +1,211 @@
+package batchrouter
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	obskit "github.com/rudderlabs/rudder-observability-kit/go/labels"
+
+	kitsync "github.com/rudderlabs/rudder-go-kit/sync"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	asynccommon "github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/common"
+	"github.com/rudderlabs/rudder-server/router/batchrouter/circuitbreaker"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+)
+
+type PartitionWorker struct {
+	wg      sync.WaitGroup
+	logger  logger.Logger
+	brt     *Handle
+	channel chan *JobEntry
+	cb      circuitbreaker.CircuitBreaker
+	limiter kitsync.Limiter
+}
+
+type JobEntry struct {
+	job      *jobsdb.JobT
+	sourceID string
+	destID   string
+}
+
+func NewPartitionWorker(logger logger.Logger, partition string, brt *Handle, cb circuitbreaker.CircuitBreaker) *PartitionWorker {
+	maxJobsToBuffer := brt.conf.GetIntVar(100000, 1, "BatchRouter."+brt.destType+".partitionWorker.maxJobsToBuffer", "BatchRouter.partitionWorker.maxJobsToBuffer")
+	pw := &PartitionWorker{
+		channel: make(chan *JobEntry, maxJobsToBuffer),
+		logger:  logger.With("partition", partition),
+		brt:     brt,
+		cb:      cb,
+		limiter: kitsync.NewReloadableLimiter(
+			context.Background(), &sync.WaitGroup{}, "batchrouter_partition_worker",
+			brt.conf.GetReloadableIntVar(10, 1, "BatchRouter."+brt.destType+".partitionWorker.concurrency", "BatchRouter.partitionWorker.concurrency"),
+			stats.Default,
+		),
+	}
+	return pw
+}
+
+func (pw *PartitionWorker) AddJob(job *jobsdb.JobT, sourceID, destID string) {
+	pw.channel <- &JobEntry{
```

Comment:
We need to capture delays here, e.g. start a goroutine that will emit a timer stat every 5 seconds with the cumulative delay